### PR TITLE
Do not depend on unversioned python binary

### PR DESCRIPTION
**Pull request by MatzeB**
Some linux distributions no longer provide `python` binary and require usage of `python3` instead. This changes the scripts here and uses cmake `find_package(Python3` when running python.

### CODE REVIEW

**Comment by LebedevRI** [comment]
Is the CMake change nessesary given the rest of the diff?
Put differently, would dropping it still solve your issue?

_Originally posted by @LebedevRI in https://github.com/google/benchmark/issues/1495#issuecomment-1265896845_

**Comment by MatzeB** [comment]
> Is the CMake change nessesary given the rest of the diff? Put differently, would dropping it still solve your issue?

Well either the cmake changes or the she-bang changes fixes things for me on CentOS8. I thought going through `find_package(Python3` was just good cmake style, but happy to drop it from the change.

**Comment by LebedevRI** [comment]
I think it's not right to have the shebang **AND** do that CMake dance.
I would guess the shebang thing should be enough,
but then we indeed probably want to check that python interpreter is present at CMake time.
But then if we run those scripts via interpreter, does the shebang help with anything?

Anyways let's just merge as is i guess.

**Comment by LebedevRI** [comment]
(please resolve the CLA issue, though.)

File: CMakeLists.txt [hunk 1]
```diff
@@ -332,6 +332,7 @@ include_directories(${PROJECT_SOURCE_DIR}/include)
 add_subdirectory(src)
 
 if (BENCHMARK_ENABLE_TESTING)
+  find_package(Python3 3.6 REQUIRED COMPONENTS Interpreter)
```
Comment (dominichamon) [inline CMakeLists.txt:335]: i read a bit about find_package but couldn't find my answer: i know this isn't an EXACT match but what versions of python3 will this match? should it be a version range?

**Comment by LebedevRI** [comment]
CLA still seems stuck, should have it unstuck?

**Comment by dominichamon** [comment]
i checked manually and @MatzeB hasn't signed the Google CLA as far as i can tell.

**Comment by MatzeB** [comment]
I am told CLA should work now. And when I go to `https://cla.developers.google.com/clas` I see "You are covered by the following corporate agreements:" "

Google Corporate CLA | Facebook, Inc. | Apr 29, 2016 12:42 PDT

[Google Corporate CLA](https://cla.developers.google.com/about/google-corporate)	Facebook, Inc.	Apr 29, 2016 12:42 PDT


Not sure why it's not working, does it still require the old xxx@fb.com mails instead of xxx@meta.com?

**Comment by LebedevRI** [comment]
Let's see if reopening PR helps.

**Comment by dominichamon** [comment]
i just forced a rescan on my end

**Review by dominichamon — APPROVED** [review]

**Comment by joshmcorreia** [comment]
This merge forced my CI/CD pipelines that use googlebenchmark to fail because our images don't have Python installed, even though googlebenchmark has never required Python before.

**Comment by LebedevRI** [comment]
> This merge forced my CI/CD pipelines that use googlebenchmark to fail because our images don't have Python installed, even though googlebenchmark has never required Python before.

Looks like you need to specify `-DBENCHMARK_ENABLE_TESTING=OFF`.

**Comment by joshmcorreia** [comment]
> Looks like you need to specify `-DBENCHMARK_ENABLE_TESTING=OFF`.

We want it on, and have had no issues with it in the past...

**Comment by LebedevRI** [comment]
Then i guess perhaps this change should be changed to use `find_program()` instead of requiring `python3-dev` package to be present, that would profile cmake files.

**Comment by LebedevRI** [comment]
> That would be great, this is currently blocking all of my company's pipelines so we can't release anything. If you could revert this change while coming up with an update it would be greatly appreciated.

I would love to, but apparently i can not.

**Comment by joshmcorreia** [comment]
Thanks for the revert!

**Pull request merged**